### ISSUE TITLE
Support for SMBv2 and up

### ISSUE THREAD

**Issue by TimOliver** [TimOliver/TOSMBClient#60]
After testing this library against various versions of SMB on my Synology NAS, it appears that libdsm only supports SMBv1 at present.

This is not great since SMBv1 has a lot of security issues and Microsoft recommends turning it off.

Unfortunately, SMBv2 is drastically different to SMBv1, so I’m not even sure if libdsm can reastically have it added as a feature.

I’m going to be researching this as much as I can. It sure is a shame that Samba is licensed under the GPL, making it incompatible on iOS. :(

**Comment by amosavian** [comment]
check please https://github.com/sahlberg/libsmb2

**Comment by amosavian** [comment]
Check https://github.com/sahlberg/libsmb2

Furthermore I would appreciate if you help me on implementing SMB2 client in my library :) https://github.com/amosavian/FileProvider

**Comment by TimOliver** [comment]
Way ahead of you: https://github.com/sahlberg/libsmb2/issues/20 :)

Hah! I'm actually working on a somewhat similar project: https://github.com/TimOliver/TOFileKit
Hopefully we can share code. :)

**Comment by amosavian** [comment]
Yes I missed that issue. I've tried to build it successfully for iOS with some changes in codes. But I'm novice on Objective C and my wrapper is getting crash on share connect.

Honestly I couldn't find the purpose of `TOFileKit`. But as I see it is relying on another old project for FTP. I'm trying to implement these protocols from scratch. Indeed I tried to implement a SMBv2 pure Swift client with MIT license. But my FTP implementation is working now.

I had problem to write an Objective-C shim for my project, especially to import protocols and their extension. It would be great if you help me on this and I can add you as a collaborator if you are interested, or move project to a new organization account.

**Comment by amosavian** [comment]
@TimOliver Find my wrapper
https://github.com/amosavian/AMSMB2

**Comment by TimOliver** [comment]
Sorry for the late reply! Oh! Congrats on building the wrapper! I'll have to give it a try myself. I've sadly not had a lot of free time lately. Since I need an Objective-C version for my projects, I might try building the library and building a `TOSMB2Client` around it.

Ah yeah, `TOFileKit` is still pretty basic at the moment. But the goal is to basically be a complete 'drop-in' solution for apps that rely on getting their files from external servers. That includes a full sign-in service for cloud accounts, (And then saving the users accounts), file browsing, and downloading. Sadly, I'm pretty old-school, so I'm writing it in Objective-C. XD

Oh yeah? I don't really have a lot of free time but I'd be happy to take a look at any Objective-C integration problems you have. :)

**Comment by amosavian** [comment]
Don't worry, you can import this framework in a ObjC project. `AMSMB2` class will be imported into Objective C. I didn't marked other classes with `@objc` attribute as they are internal.

I'm looking forward for any possible cooperation. :D
